src/components/Tabs/index.js

```diff
@@ -0,0 +1,223 @@
+
+  &.overflowed {
+    margin-left: calc(50% - 50vw);
+    padding-left: calc(50vw - 50%);
+    width: 100vw;
+  }
+
+  &::-webkit-scrollbar {
+    display: none;
+  }
+`;
+
+const Item = styled.div`
+  ${largeAndUp`
+    padding-right: ${spacing.comfy};
+  `};
+  display: block;
+  padding-right: ${spacing.normal};
+  white-space: nowrap;
+
+  &:last-of-type {
+    padding-right: 0;
+  }
+`;
+
+const checkIfOverflows = ({ offsetWidth = 0, scrollWidth = 0 } = {}) =>
+  offsetWidth < scrollWidth;
+
+const TabItemButton = styled.div.attrs({
+  className: props => (props.isActive ? "isActive" : ""),
+  "data-index": props => props.dataIndex
+})`
+  display: block;
+  padding: ${padding} 0 ${spacing.cozy} 0;
+  cursor: pointer;
+  &.isActive {
+    border-bottom: 4px solid
+      ${props => props.underlineColor || colors.azure.base};
+  }
+`;
+
+class Tabs extends Component {
+  static propTypes = {
+    items: PropTypes.arrayOf(PropTypes.node).isRequired,
+    index: PropTypes.number,
+    headingIndex: PropTypes.number,
+    onClick: PropTypes.func,
+    variant: PropTypes.string,
+    accent: PropTypes.string,
+    weight: PropTypes.string,
+    underlineColor: PropTypes.string
+  };
+
+  static defaultProps = {
+    index: -1,
+    headingIndex: -1,
+    onClick: () => {},
+    variant: null,
+    accent: null,
+    weight: null,
+    underlineColor: null
+  };
+
+  state = {
+    isTabsContainerOverflows: false
+  };
+
+  componentDidMount() {
+    this.updateIsOverflowsState();
+    window.addEventListener("resize", this.updateIsOverflowsState); // eslint-disable-line
+  }
+
+  componentDidUpdate() {
+    if (this.state.isTabsContainerOverflows) {
+      const {
+        offsetWidth: contentWidth,
+        scrollLeft: contentScrollLeft
+      } = this.content;
+      const { index } = this.props;
+      const {
+        offsetWidth: activeItemWidth = 0,
+        offsetLeft: activeItemLeft = 0
+      } = this.itemRefs[index];
+
+      if (activeItemWidth + activeItemLeft > contentWidth + contentScrollLeft) {
+        this.content.scrollLeft =
+          activeItemWidth +
+          activeItemLeft -
+          contentWidth +
+          parseInt(padding, 10);
+      } else if (activeItemLeft < contentScrollLeft) {
+        this.content.scrollLeft = activeItemLeft - parseInt(padding, 10);
+      }
+    }
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("resize", this.updateIsOverflowsState); //eslint-disable-line
+  }
+
+  updateIsOverflowsState = () => {
+    const isOverflows = checkIfOverflows(this.content);
+    if (this.state.isTabsContainerOverflows !== isOverflows) {
+      this.setState({
+        isTabsContainerOverflows: isOverflows
+      });
+    }
+  };
+
+  itemRefs = [];
+
+  renderItem = ({ itemIndex, item, ...itemprops }) => {
+    const { index, onClick, underlineColor } = this.props;
+    return (
+      <TabItemButton
+        isActive={itemIndex === index}
+        underlineColor={underlineColor}
```

Comment:
What's the use case for this prop?

Reply:
This could be applied if we need different underline color for selected tab. It is optional and default color is colors.azure.base.